belt.go

```diff
@@ -102,7 +104,7 @@ func (belt *Belt) Fields() field.AbstractFields {
 func (belt *Belt) WithTool(toolID ToolID, tool Tool) *Belt {
 	clone := belt.clone()
 
-	tools := make(Tools, len(clone.tools))
+	tools := make(Tools, len(clone.tools)+1)
```

Comment:
Just a small optimization, feel free to push back on this, since the change is not related to this PR.